atomic_reactor/inner.py

```diff
@@ -414,6 +416,14 @@ def build_docker_image(self):
                 logger.error("one or more prepublish plugins failed: %s", ex)
                 raise
 
+            if self.build_result.is_image_available():
+                history = self.builder.tasker.d.history(self.builder.image_id)
+                diff_ids = self.built_image_inspect[INSPECT_ROOTFS]['Layers']
+                diff_ids.reverse()
+                for layer_idx in range(len(history)):
+                    self.built_image_layer_sizes.insert(0, {diff_ids[layer_idx]:
+                                                                history[layer_idx]['Size']})
```

Comment:
This might be easier to follow as a list comprehension, and definitely needs comments explaining which order things start off in.

e.g. (not sure if I have the order right)
```
# diff_ids is ordered oldest first
# history is ordered newest first
# We want layer_sizes to be ordered oldest first
layer_sizes = [{diff_id: layer['Size']}
               for (diff_id, layer) in zip(diff_ids, reversed(history))]
```